scripts/test/Muon/grouping_tab/pairing_table_alpha_test.py

```diff
@@ -0,0 +1,146 @@
+        group3 = MuonGroup(group_name="my_group_2", detector_ids=[3])
+        self.data.add_group(group1)
+        self.data.add_group(group2)
+        self.data.add_group(group3)
+
+    def add_two_pairs_to_table(self):
+        pair1 = MuonPair(pair_name="my_pair_0", forward_group_name="my_group_0", backward_group_name="my_group_1", alpha=1.0)
+        pair2 = MuonPair(pair_name="my_pair_1", forward_group_name="my_group_1", backward_group_name="my_group_2", alpha=1.0)
+        self.presenter.add_pair(pair1)
+        self.presenter.add_pair(pair2)
+
+    def get_group_1_selector(self, row):
+        return self.view.pairing_table.cellWidget(row, 1)
+
+    def get_group_2_selector(self, row):
+        return self.view.pairing_table.cellWidget(row, 2)
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # TESTS : test the functionality around alpha.
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def test_that_alpha_defaults_to_1(self):
+        self.presenter.handle_add_pair_button_clicked()
+
+        self.assertEqual(self.view.get_table_item_text(0, 3), "1.0")
+
+    def test_that_table_reverts_to_previous_value_when_adding_values_which_arent_numbers_to_alpha_column(self):
+        self.presenter.handle_add_pair_button_clicked()
+
+        non_numeric_alphas = ["", "a", "long", "!", "_", "1+2"]
+
+        default_value = self.view.get_table_item_text(0, 3)
+        for invalid_alpha in non_numeric_alphas:
+            self.view.pairing_table.setCurrentCell(0, 3)
+            self.view.pairing_table.item(0, 3).setText(invalid_alpha)
+
+            self.assertEqual(self.view.get_table_item_text(0, 3), default_value)
+
+    def test_that_warning_displayed_when_adding_invalid_alpha_values(self):
+        self.presenter.handle_add_pair_button_clicked()
+
+        non_numeric_alphas = ["", "a", "long", "!", "_", "1+2"]
+
+        call_count = 0
+        for invalid_alpha in non_numeric_alphas:
+            call_count += 1
+            self.view.pairing_table.setCurrentCell(0, 3)
+            self.view.pairing_table.item(0, 3).setText(invalid_alpha)
+
+            self.assertEqual(self.view.warning_popup.call_count, call_count)
+
+    def test_that_alpha_values_stored_to_three_decimal_places(self):
+        self.presenter.handle_add_pair_button_clicked()
+
+        self.view.pairing_table.setCurrentCell(0, 3)
+        # test that rounds correctly
+        self.view.pairing_table.item(0, 3).setText("1.1239")
+
+        self.assertEqual(self.view.get_table_item_text(0, 3), "1.124")
+
+    def test_that_valid_alpha_values_are_added_correctly(self):
+        self.presenter.handle_add_pair_button_clicked()
+
+        valid_inputs = ["1.0", "0.0", "12", ".123", "0.00001", "0.0005"]
+        expected_output = ["1.0", "0.0", "12.0", "0.123", "0.0", "0.001"]
```

Comment:
the 0.0 in the expected output makes me feel uneasy as it is misleading as alpha is not zero. Maybe we could give alpha in standard form if it is below 3 dp or just not allow it?